frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -438,6 +470,169 @@ pub mod pallet {
+        /// If unstaking the specified `value` would take staker below the `T::MinimumStakingAmount`,
+        /// entire staked amount will be unstaked.
+        ///
+        /// If user attempts to unstake more than is available, everything will be unstaked.
+        ///
+        /// The dispatch origin for this call must be _Signed_ by the staker's account.
+        ///
+        /// Note that all funds will remain staked after this call, only the unbonding process is started.
+        /// Staker will still earn rewards as long as funds aren't withdrawn.
+        ///
+        #[pallet::weight(1000_0000)]
+        pub fn start_unbonding(
+            origin: OriginFor<T>,
+            contract_id: T::SmartContract,
+            #[pallet::compact] value: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let staker = ensure_signed(origin)?;
+
+            ensure!(value > Zero::zero(), Error::<T>::UnstakingWithNoValue);
+            ensure!(
+                Self::is_active(&contract_id),
+                Error::<T>::NotOperatedContract,
+            );
+
+            // Get the unlocking chunks for this staker and contract
+            let mut unbonding_info = UnbondingInfoStorage::<T>::get(&staker, &contract_id);
+            ensure!(
+                unbonding_info.len() < T::MaxUnlockingChunks::get(),
+                Error::<T>::TooManyUnlockingChunks
+            );
+            let unlocking_chunks_sum = unbonding_info.sum();
+
+            // Get the latest era staking points for the contract.
+            let current_era = Self::current_era();
+            let staking_info = Self::staking_info(&contract_id, current_era);
+            ensure!(
+                staking_info.stakers.contains_key(&staker),
+                Error::<T>::NotStakedContract,
+            );
+            let staked_value = staking_info.stakers[&staker];
+
+            // Calculate the value which can be unstaked.
+            let value_available_for_unstaking = staked_value.saturating_sub(unlocking_chunks_sum);
+            let remaining_value_after_unstaking =
+                value_available_for_unstaking.saturating_sub(value);
+
+            // In case we would go below the minimum staking amount, unstake everything
+            let value_to_unstake =
+                if remaining_value_after_unstaking < T::MinimumStakingAmount::get() {
+                    value_available_for_unstaking
+                } else {
+                    value.min(value_available_for_unstaking)
+                };
+
+            // It is possible that the entire value is already in the process of being unstaked.
+            ensure!(
+                value_to_unstake > Zero::zero(),
+                Error::<T>::UnstakingWithNoValue
+            );
+
+            // TODO: NOTE that UnbondingPeriod doesn't mean the full eras. E.g. if it is set to 1, it will expire after current era.
```

Comment:
Please comment this.
I am inclined to adding +1 here.

Reply:
I feel +1 would be good. 
perhaps @bLd75 and @fiexer can comment

Reply:
So +1 is unbond after end next era?

Reply:
+1 means that we don't consider the current era into the calculation. We are only interested in the full eras.

**Example:**
unbonding_period = 3;
current_era = 10;

**_1st  scenario_**
No +1.
User calls start_unbonding and unbonding process begins.
unlock_era = current_era + unbonding_period = 13
This means that in era 13, user can unstake & withdraw the unbonded funds.

Note that we don't have a guarantee how much of era 10 contributed to the unbonding period.
Perhaps the start_unbonding was called in the last block of the era.

**_2nd scenario_**
+1.
unlock_era = current_era + unbonding_period + 1 = 14

Now we have a guarantee that user will have to wait _unbonding_period_ number of FULL eras before funds
become available for unstake & withdraw.


Hope the examples help.